configs/components/fesom/fesom-2.1.yaml

```diff
@@ -256,27 +250,61 @@ choose_use_icebergs:
-                    length: "icb_length.dat"
-                    height: "icb_height.dat"
-                    scaling: "icb_scaling.dat"
-
-            input_sources:
-                    longitude: "${iceberg_dir}/LON.dat"
-                    latitude: "${iceberg_dir}/LAT.dat"
-                    length: "${iceberg_dir}/LENGTH.dat"
-                    height: "${iceberg_dir}/HEIGHT.dat"
-                    scaling: "${iceberg_dir}/SCALING.dat"
+                            add_input_files:
+                                    latitude: "latitude"
+                                    longitude: "longitude"
+                                    length: "length"
+                                    height: "height"
+                                    scaling: "scaling"
+                                    felem: "felem"
+                                    num_non_melted_icb_file: "num_non_melted_icb_file"
+                    false:
+                            add_input_sources:
+                                    length:    "${iceberg_dir}/icb_length.dat_@YEAR@0101-@YEAR@1231"
+                                    latitude:  "${iceberg_dir}/icb_latitude.dat_@YEAR@0101-@YEAR@1231"
+                                    longitude: "${iceberg_dir}/icb_longitude.dat_@YEAR@0101-@YEAR@1231"
+                                    height:    "${iceberg_dir}/icb_height.dat_@YEAR@0101-@YEAR@1231"
+                                    scaling:   "${iceberg_dir}/icb_scaling.dat_@YEAR@0101-@YEAR@1231"
+                                    felem:   "${iceberg_dir}/icb_felem.dat_@YEAR@0101-@YEAR@1231"
+                                    num_non_melted_icb_file: "${iceberg_dir}/num_non_melted_icb_file_@YEAR@0101-@YEAR@1231"
+            
+                            input_in_work:
+                                    latitude: "icb_latitude.dat"
+                                    longitude: "icb_longitude.dat"
+                                    length: "icb_length.dat"
+                                    height: "icb_height.dat"
+                                    scaling: "icb_scaling.dat"
+                                    felem: "icb_felem.dat"
+                                    num_non_melted_icb_file: "num_non_melted_icb_file"
 
-            input_files:
-                    longitude: "longitude"
-                    latitude: "latitude"
-                    length: "length"
-                    height: "height"
-                    scaling: "scaling"
+                            add_input_files:
+                                    latitude: "latitude"
+                                    longitude: "longitude"
+                                    length: "length"
+                                    height: "height"
+                                    scaling: "scaling"
+                                    felem: "felem"
+                                    num_non_melted_icb_file: "num_non_melted_icb_file"
```

Comment:
There is a lot of repetition here. Have you considered using the multilist (list loops) feature?

https://esm-tools.readthedocs.io/en/latest/yaml.html#list-loops

Let me know if you need help with that.